test/linux/mbox-server-tester.c

```diff
@@ -0,0 +1,132 @@
+#define _GNU_SOURCE
+
+#include <errno.h>
+#include <fcntl.h>
+#include <getopt.h>
+#include <sched.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "libmbox/mbox.h"
+#include "mailbox-map.h"
+#include "mbox-utils.h"
+
+#define PATH_SIZE 128
+
+#define RTPS_PATH_IN "/dev/mbox/1/mbox4"
+#define RTPS_PATH_OUT "/dev/mbox/1/mbox3"
+
+
+enum mbox_notif notif_type = MBOX_NOTIF_NONE;
+int timeout_ms = -1;
+
+static const char short_options[] = "n:t:c:h";
+static const struct option long_options[] = {
+    {"not-type",    required_argument,  NULL,   'n'},
+    {"timeout",     required_argument,  NULL,   't'},
+    {"cpu",         required_argument,  NULL,   'c'},
+    {"help",        no_argument,        NULL,   'h'},
+    {0, 0, 0, 0}
+};
+
+
+static void usage(const char *pname, int code)
+{
+    fprintf(code ? stderr : stdout,
+            "Usage: %s [-n TYPE] [-t N] [-c CPU] [-h]\n"
+            "  -n, --not-type=TYPE  Notification type, one of: none, select, poll, epoll\n"
+            "                       The default value is \"none\"\n"
+            "  -t, --timeout=N      Timeout for reads, in milliseconds\n"
+            "                       The default value is -1 (infinite timeout)\n"
+            "                       Specify a positive value, or 0 for no timeout\n"
+            "  -c, --cpu=CPU        Pin process to CPU\n"
+            "                       The default value is -1, for no pinning\n"
+            "  -h, --help           Print this message and exit\n",
+            pname);
+    exit(code);
+}
+
+int main(int argc, char** argv)
+{
+    struct mbox mbox_in;
+    struct mbox mbox_out;
+
+    int cpu = -1;
+
+    int c;	
+    while ((c = getopt_long(argc, argv, short_options, long_options, NULL)) != -1) {
+        switch (c) {
+        case 'n':
+            if (!strcmp(optarg, "none")) {
+                notif_type = MBOX_NOTIF_NONE;
+            } else if (!strcmp(optarg, "select")) {
+                notif_type = MBOX_NOTIF_SELECT;
+            } else if (!strcmp(optarg, "poll")) {
+                notif_type = MBOX_NOTIF_POLL;
+            } else if (!strcmp(optarg, "epoll")) {
+                notif_type = MBOX_NOTIF_EPOLL;
+            } else {
+                fprintf(stderr, "Unknown notification type: %s\n", optarg);
+                usage(argv[0], EINVAL);
+            }
+            break;
+        case 't':
+            timeout_ms = atoi(optarg);
+            break;
+        case 'c':
+            cpu = atoi(optarg);
+            break;
+        case 'h':
+            usage(argv[0], 0);
+            break;
+        default:
+            usage(argv[0], EINVAL);
+            break;
+    	}
+	}
+
+	notif_type = MBOX_NOTIF_POLL;
```

Comment:
This line seems to short-circuit the notification type choice made by the user so that it always polls.  I don't think this is what we want, is it?  In addition, there is a tab preceding this line.